clippy_lints/src/methods/or_fun_call.rs

```diff
@@ -52,16 +53,24 @@ pub(super) fn check<'tcx>(
 
             then {
                 let mut applicability = Applicability::MachineApplicable;
+                let hint = ".unwrap_or_default()";
+                let mut sugg: String = format!(
+                    "{}{}",
+                    snippet_with_applicability(cx, self_expr.span, "..", &mut applicability),
+                    hint
+                );
+
+                if sugg.lines().count() > MAX_SUGGESTION_HIGHLIGHT_LINES {
+                    sugg = hint.to_string();
+                }
+
                 span_lint_and_sugg(
                     cx,
                     OR_FUN_CALL,
                     span,
                     &format!("use of `{}` followed by a call to `{}`", name, path),
                     "try this",
-                    format!(
-                        "{}.unwrap_or_default()",
-                        snippet_with_applicability(cx, self_expr.span, "..", &mut applicability)
-                    ),
+                    sugg,
```

Comment:
If you change the suggestion to only display `.unwrap_or_default()` then the span in line 70 also has to change to only replace the previous function call. This would currently replace the entire expression with just the string `.unwrap_or_default()`.

I would recomment writing a test first and then checking the output. :upside_down_face: The test is also needed to ensure that this doesn't happen again :)